blog/2024-11-18-relay-ops.md

```diff
@@ -0,0 +1,60 @@
+tags: ['firehose']
+---
+
+**Summary:** we are making a couple changes to the Bluesky Relay servers **this week**. The wire protocol and semantics are not impacted, but the changes may impact firehose consumers: 
+
+- one way or another, event stream sequence will change for all subscribers. Most consumers can probably just reset to the new cursor; see details below.
+- the `bsky.network` hostname will be swapped to a new relay instance (with new sequence)
+
+The atproto networking is growing rapidly! This is exciting and positive, but means we need to move up some of our network scaling plans. One particular resource bottleneck is the Relay firehose. We are seeing sustained traffic of over 2,000 events per second, and have hundreds of active consumers. Multiplied, the overall throughput is hard for any one server to keep up with.
+
+We saw this problem coming from the beginning, and have several plans and options to mitigate it. Details for each of these are linked or in sections below.
+
+First, we are simply upgrading the size of server that we run our primary Relay instances on. We expect to do this in coming days. The main impact here will be a reset of the firehose sequence (including cursor values), which will impact all downstream consumers.
+
+Second, we are introducing the concept of "firehose fan-out" services, and releasing our implementation [Rainbow](https://github.com/bluesky-social/indigo/tree/main/cmd/rainbow). These are servers which re-broadcast the event stream firehose to many clients, reducing the bandwidth load on relays themselves. Rainbow is now in production, as an implementation detail.
+
+Third, we recently released [Jetstream](https://docs.bsky.app/blog/jetstream) as a more informal and light-weight option for event stream consumption. We encourage developers to check out Jetstream, and switch over if it works for their use case.
+
+We encourage folks to take advantage of Rainbow and Jetstream. You can run your own, or make use of our hosted instances. Some additional longer-term options are discussed at the bottom of this article.
+
+We don't have hard time or date estimate for the cursor sequence change, but expect it this week, possibly tomorrow (Tuesday). We generally try to give more time and make changes less disruptive, but are not able to do so in this situation. Please follow the [`@atproto.com`](https://bsky.app/profile/atproto.com) account, this blog, and in [Github Discussions](https://github.com/bluesky-social/atproto/discussions) for announcements.
+
+## Relay Upgrade
+
+The current Bluesky relay instance is available at `wss://bsky.network`. A more specific hostname for this relay instance is (XXX: public old relay hostname).
+
+Instead of upgrading the relay in-place, we are going to start a new relay from scratch, which means event stream ordering will be different and sequence numbers will not align with the current relay. This new instance will run at (XXX: public new relay hostname).
```

Comment:
```suggestion
Instead of upgrading the relay in-place, we are going to start a new relay from scratch. This means that event stream ordering will be different and sequence numbers will not align with the current relay. This new instance will run at (XXX: public new relay hostname).
```